app/views/payment_gateways/_stripe.html.erb

```diff
@@ -1,128 +1,22 @@
-      </div>
-    </fieldset>
-  <% end %>
+<%
+  billing_address = current_account.billing_address
+  billing_address_json = {
+    line1: billing_address.address1,
+    line2: billing_address.address2,
+    city: billing_address.city,
+    state: billing_address.state,
+    postal_code: billing_address.zip,
+    country: billing_address.country
+  }.to_json
+%>
+
+<div id="stripe-form-wrapper"
+  data-stripe-publishable-key="<%= @stripe_publishable_key %>"
+  data-setup-intent-secret="<%= @intent.client_secret %>"
+  data-billing-address="<%= billing_address_json %>"
+  data-success-url="<%= hosted_success_admin_account_stripe_path %>"
+  data-credit-card-stored="<%= current_account.credit_card_stored? %>">
 </div>
 
-<% billing_address = current_account.billing_address %>
-
-<script type="text/javascript" charset="utf-8">
-  var stripeForm = $('#stripe-form');
-
-  $('#toggle-stripe-form').click(function() {
-    stripeForm.toggleClass('hidden');
-    var linkText = stripeForm.is(':visible') ? 'cancel' : 'Edit Credit Card Details';
-    $(this).find('span').text(linkText);
-    return false;
-  });
-
-  var stripe = Stripe('<%= @stripe_publishable_key %>');
-  var elements = stripe.elements();
-
-  var card = elements.create('card', {
-    iconStyle: 'solid',
-    style: {
-      base: {
-        iconColor: '#8898AA',
-        color: 'black',
-        fontWeight: 300,
-        fontFamily: '"Helvetica Neue", Helvetica, sans-serif',
-        fontSize: '19px',
-        '::placeholder': {
-          color: '#3f454c',
-        },
-      },
-      invalid: {
-        iconColor: '#e85746',
-        color: '#e85746',
-      }
-    },
-    classes: {
-      focus: 'is-focused',
-      empty: 'is-empty',
-    }
-  });
-
-  var submit = $('#stripe-submit');
-  card.on('change', function (e) {
-    submit.prop('disabled', !e.complete)
-  });
-
-  card.mount('#card-element');
-
-  function handleIntentResult(result) {
-    if (result.setupIntent && result.setupIntent.status === "succeeded") {
-      var setupIntent = result.setupIntent;
-
-      document.getElementById('stripe_payment_method_id').value = setupIntent.payment_method;
-
-      stripeForm[0].submit();
-    } else {
-      alert(result.error.message);
-    }
-  }
-
-  stripeForm.submit(function(e) {
-    e.preventDefault();
-    var intentSecret = stripeForm.data('secret');
-
-    var billingAddressDetails = {
-      line1: "<%= j billing_address.address1 %>",
-      line2: "<%= j billing_address.address2 %>",
-      city: "<%= j billing_address.city %>",
-      state: "<%= j billing_address.state %>",
-      postal_code: "<%= j billing_address.zip %>",
-      country: "<%= j billing_address.country %>"
-    }
-
-    stripe.confirmCardSetup(intentSecret, {
-      payment_method: {
-        card: card,
-        billing_details: {
-          address: billingAddressDetails
-        }
-      }
-    }).then(handleIntentResult);
-  });
-
-</script>
+<script src="https://js.stripe.com/v3/"></script>
```

Comment:
Shouldn't be needed since we're using the NPM package.
```suggestion
```

Reply:
@Martouta @josemigallas 
I will leave this script, and also include it in the main layout
Newly created templates will load it from main layout, but we need to keep this in the partial to make it work with already created templates. It's redundant, but I don't see other workaround